lnd_test.go

```diff
@@ -3032,6 +3088,199 @@ func testPrivateChannels(net *lntest.NetworkHarness, t *harnessTest) {
 	}
 }
 
+// testMultiHopOverPrivateChannels tests that private channels can be used as
+// intermediate hops in a route for payments.
+func testMultiHopOverPrivateChannels(net *lntest.NetworkHarness, t *harnessTest) {
+	// We'll test that multi-hop payments over private channels work as
+	// intended. To do so, we'll create the following topology:
+	//         public         private          private
+	// Alice <--100k--> Bob <--100k--> Carol <--100k--> Dave
```

Comment:
Related to my earlier comments in the routing package, I'm not sure this is actually yet achievable within the constraints of the protocol. How would Dave obtain information concerning the Bob <-> Carol private channel? Instead, I'd expect the topology to be something like: 
```
//      private                public                  private
// Alice <--100k--> Bob <--100k--> Carol <--100k--> Dave
```

Reply:
The spec is a bit vague, but from what I read you can provide multiple ordered channels (a route), not just single last hops. I always assumed this was to allow cases like for instance Dave being a good friend of Bob, so Bob has shared his channels with him (or maybe Dave can purchase knowledge about private channels from a provider?).

Reply:
Modified the test case to follow the model above.

Something like
```
//         public        private          private
// Alice <--100k--> Bob <--100k--> Carol <--100k--> Dave
```
is still possible with the current implementation as long as Dave knows about the channel between Bob and Carol. I can add another test case for that if needed.